### PR TITLE
Some clang tidy fixes from Google weekly import

### PR DESCRIPTION
**Pull request by hennna**
Signed-off-by: Henna Huang <henna@google.com>

*title*: *Some clang tidy fixes from Google weekly import*

*Description*: Clang tidy fixes
  - remove unused using
  - match header and function definition parameter names
  - s/size() > 0/empty()
  - const Address::InstanceConstSharedPtr

*Risk Level*: Low

*Testing*: no additional tests added

*Docs Changes*: none

*Release Notes*: none

### CODE REVIEW

File: source/common/network/connection_impl.cc [hunk 1]
```diff
@@ -534,11 +534,11 @@ void ConnectionImpl::updateWriteBufferStats(uint64_t num_written, uint64_t new_s
                                            connection_stats_->write_current_);
 }
 
-ClientConnectionImpl::ClientConnectionImpl(
-    Event::Dispatcher& dispatcher, Address::InstanceConstSharedPtr address,
-    const Network::Address::InstanceConstSharedPtr source_address)
-    : ConnectionImpl(dispatcher, address->socket(Address::SocketType::Stream), address, nullptr,
-                     source_address, false, false) {}
+ClientConnectionImpl::ClientConnectionImpl(Event::Dispatcher& dispatcher,
+                                           const Address::InstanceConstSharedPtr remote_address,
```
Comment (htuch) [inline source/common/network/connection_impl.cc]: Is this really what we want here? Is the idea that the reference count never changes? If so, why don't we just pass in a const reference to the address instance and avoid the involvement of shared_ptr?

Comment (mattklein123) [reply source/common/network/connection_impl.cc]: I think all of these are just "bugs". They should be const&. Same below. I would change them.

File: source/common/network/connection_impl.h [hunk 2]
```diff
@@ -172,8 +172,8 @@ class ConnectionImpl : public virtual Connection,
 class ClientConnectionImpl : public ConnectionImpl, virtual public ClientConnection {
 public:
   ClientConnectionImpl(Event::Dispatcher& dispatcher,
-                       const Address::InstanceConstSharedPtr remote_address,
-                       const Address::InstanceConstSharedPtr source_address);
+                       const Address::InstanceConstSharedPtr& remote_address,
```
Comment (htuch) [inline source/common/network/connection_impl.h]: I think "Guideline: Don’t use a const unique_ptr& as a parameter; use widget* instead." at https://herbsutter.com/2013/06/05/gotw-91-solution-smart-pointer-parameters/ might apply here.

Comment (mattklein123) [reply source/common/network/connection_impl.h]: Where is there a unique_ptr? These are all shared, and we store them in the connection so they have to be passed as such?

Comment (htuch) [reply source/common/network/connection_impl.h]: Sorry, it was the stuff further down from that in that same guideline section. In any case, looking further at the code, we do want to take a shared copy later, just not temporarily here, so what is there is fine actually.

**Pull request merged**